include/dxc/Support/HLSLOptions.td

```diff
@@ -300,6 +300,8 @@ def disable_lifetime_markers : Flag<["-", "/"], "disable-lifetime-markers">, Gro
   HelpText<"Disable generation of lifetime markers where they would be otherwise (6.6+)">;
 def enable_templates: Flag<["-", "/"], "enable-templates">, Group<hlslcomp_Group>, Flags<[CoreOption, HelpHidden]>,
   HelpText<"Enable template support for HLSL.">;
+def enable_unions: Flag<["-", "/"], "enable-unions">, Group<hlslcomp_Group>, Flags<[CoreOption]>,
```

Comment:
@pow2clk or @tex3d may have a different opinion from me, but I'd prefer if we didn't add individual feature flags for HLSL 202x features. I think they complicate the testing configurations, and we won't want to support them on release.

Reply:
We still have this flag in the PR, but enable unions by default with LangStd::v202x. We can remove the flag entirely if you'd still like.

Reply:
We should remove the option entirely, and add test cases to verify that use of unions causes an error with `-HV 2021`.

I think the code still enables unions under HLSL 2021 because the comparison is `>= 2021`, and should probably be `>= 202x`.